tests/parsers/test.manifestjson.js

```diff
@@ -808,125 +806,136 @@ describe('ManifestJSONParser', () => {
+          },
+        });
+        const files = {
+          [icon32]: fs.createReadStream(icon32),
+          [icon2048]: fs.createReadStream(icon2048),
+        };
+
+        const manifestJSONParser = new ManifestJSONParser(
+          json, addonLinter.collector, { io: getStreamableIO(files) });
+
+        await Promise.all([
+          manifestJSONParser.validateIcon(icon32, size32),
+          manifestJSONParser.validateIcon(icon2048, size2048),
+        ]);
+        expect(manifestJSONParser.isValid).toBeTruthy();
+        const { warnings } = addonLinter.collector;
+        expect(warnings.length).toEqual(0);
       });
-      const files = {
-        'tests/fixtures/default.png': fs.createReadStream('tests/fixtures/default.png'),
-        'tests/fixtures/default.svg': fs.createReadStream('tests/fixtures/default.svg'),
-      };
 
-      const manifestJSONParser = new ManifestJSONParser(
-        json, addonLinter.collector, { io: getStreamableIO(files) });
-      return manifestJSONParser.validateIcons()
-        .then(() => {
-          expect(manifestJSONParser.isValid).toBeTruthy();
-          const { warnings } = addonLinter.collector;
-          expect(warnings.length).toEqual(0);
+      it('adds a warning if the icon file is corrupt', async () => {
+        const addonLinter = new Linter({ _: ['bar'] });
+        const icon32 = 'tests/fixtures/default-corrupted.png';
+        const size32 = 32;
+        const json = validManifestJSON({
+          icons: {
+            [size32]: icon32,
+          },
         });
-    });
+        const files = {
+          [icon32]: fs.createReadStream(icon32),
+        };
 
-    it('adds a warning if the icon file is corrupt', () => {
-      const addonLinter = new Linter({ _: ['bar'] });
-      const json = validManifestJSON({
-        icons: {
-          32: 'tests/fixtures/default-corrupted.png',
-        },
+        const manifestJSONParser = new ManifestJSONParser(
+          json, addonLinter.collector, { io: getStreamableIO(files) });
+
+        await manifestJSONParser.validateIcon(icon32, size32);
+        expect(manifestJSONParser.isValid).toBeTruthy();
+        const { warnings } = addonLinter.collector;
+        expect(warnings.length).toEqual(1);
       });
-      const files = {
-        'tests/fixtures/default-corrupted.png': fs.createReadStream('tests/fixtures/default-corrupted.png'),
-      };
 
-      const manifestJSONParser = new ManifestJSONParser(
-        json, addonLinter.collector, { io: getStreamableIO(files) });
-      return manifestJSONParser.validateIcons()
-        .then(() => {
-          expect(manifestJSONParser.isValid).toBeTruthy();
-          const { warnings } = addonLinter.collector;
-          expect(warnings.length).toEqual(1);
+      it('adds a warning if the image size is not the same as mentioned', async () => {
+        const addonLinter = new Linter({ _: ['bar'] });
+        const icon32 = 'tests/fixtures/icon-33.png';
```

Comment:
Nit, how about rename this to `icon33`? so that the size mismatch is also visible when looking at the identifiers names.